web3-functions/index.ts

```diff
@@ -0,0 +1,293 @@
+import {
+  Web3Function,
+  Web3FunctionContext,
+} from "@gelatonetwork/web3-functions-sdk";
+import { BigNumber, Contract } from "ethers";
+import { arrayify, toUtf8String, formatBytes32String } from "ethers/lib/utils";
+import { WrapperBuilder } from "@redstone-finance/evm-connector";
+import * as redstone from "redstone-protocol";
+
+const ORACLE_ABI = [
+  "function updateDataFeedsValuesPartial(bytes32[]) public",
+  "function getLastUpdateDetails(bytes32) public view returns (uint256, uint256, uint256)",
+  "function getLivePricesAndTimestamp(bytes32[]) public view returns (uint256[], uint256)",
+];
+
+const MIN_DEVIATION = 0.5; // 0.5%
+const MIN_TIME_ELAPSED = 6; // 6 hours
+const DECIMALS = 8; // price feed precision
+const zero = BigNumber.from(0);
+const one = BigNumber.from(1);
+
+const DEBUG_MODE = false;
+const debugLog = conditionalLog(DEBUG_MODE);
+
+interface DataFeed {
+  symbol: string;
+  id: string;
+  livePrice: BigNumber;
+  storedPrice: BigNumber;
+  storedTimestamp: number;
+}
+
+Web3Function.onRun(async (context: Web3FunctionContext) => {
+  const { userArgs, multiChainProvider } = context;
+
+  const provider = multiChainProvider.default();
+
+  const dataServiceId = userArgs.dataServiceId as string;
+  const dataFeedIdsString = userArgs.symbols as string[];
+  const dataFeedIdsBytes32 = dataFeedIdsString.map((id) =>
+    formatBytes32String(id)
+  );
+  const oracleAddress = userArgs.oracleAddress as string;
+
+  const oracle = new Contract(oracleAddress, ORACLE_ABI, provider);
+
+  let dataFeedIds = new Map<string, DataFeed>();
```

Comment:
For clarity.
```suggestion
  const idToDataFeed = new Map<string, DataFeed>();
```
or

```suggestion
  const idToDataFeedMap = new Map<string, DataFeed>();
```

Reply:
Will be addressed in separate [Issue](https://github.com/LiskHQ/lisk-contracts/issues/210)